### PR TITLE
Add link column to csv file

### PR DESCRIPTION
**Pull request by SumeN**
Addition of View AATF Data links in the CSV

### CODE REVIEW

File: src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs [hunk 1]
```diff
@@ -477,8 +477,8 @@ public async Task<ActionResult> AatfAeReturnData(AatfAeReturnDataViewModel model
           int quarter,  FacilityType facilityType, int? submissionStatus, Guid? authority, Guid? pat, Guid? localArea)
         {
             CSVFileData fileData;
-
-            GetAatfAeReturnDataCsv request = new GetAatfAeReturnDataCsv(complianceYear, quarter, facilityType, submissionStatus, authority, pat, localArea);
+            string aatfDataUrl = HttpContext.Request.Url != null ? string.Concat(HttpContext.Request.Url.Scheme, "://", HttpContext.Request.Url.Authority, "/admin/aatf/details/") : string.Empty;
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs]: We should be able to mock the HTTP context using HttpContextMocker so you can test that the request is passed the the correct URL based on the mocked httpcontext

File: src/EA.Weee.RequestHandlers/Admin/Reports/GetAatfAeReturnDataCsvHandler.cs [hunk 2]
```diff
@@ -37,6 +37,11 @@ public async Task<CSVFileData> HandleAsync(GetAatfAeReturnDataCsv request)
                        request.ComplianceYear, request.Quarter, (int)request.FacilityType, 
                        request.ReturnStatus, request.AuthorityId, request.LocalArea, request.PanArea);
 
+            foreach (var item in items)
+            {
+                 item.AatfDataUrl = string.Format(@" =HYPERLINK(""""{0}{1}#data"""", """"View AATF data"""")", request.AatfDataUrl, item.AatfId);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.RequestHandlers/Admin/Reports/GetAatfAeReturnDataCsvHandler.cs]: Just add test to check that this hyperlink is set correctly

File: src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs (hunk 1 above)
Comment (SumeN) [reply src/EA.Weee.Web/Areas/Admin/Controllers/ReportsController.cs]: Added

File: src/EA.Weee.RequestHandlers/Admin/Reports/GetAatfAeReturnDataCsvHandler.cs (hunk 2 above)
Comment (SumeN) [reply src/EA.Weee.RequestHandlers/Admin/Reports/GetAatfAeReturnDataCsvHandler.cs]: Added

File: src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ReportsControllerTests.cs [hunk 3]
```diff
@@ -1816,5 +1819,39 @@ public async Task PostAatfAeReturnData_Always_SetsInternalBreadcrumbToViewReport
...
+            A.CallTo(() => apiClient.SendAsync(A<string>._, A<GetAatfAeReturnDataCsv>._)).Returns(file);
+
+            ActionResult result = await controller.DownloadAatfAeDataCsv(2019, 1,
+                FacilityType.Aatf, A.Dummy<int?>(), A.Dummy<Guid?>(), A.Dummy<Guid?>(), 
+                A.Dummy<Guid?>());
+
+            A.CallTo(() => apiClient.SendAsync(A<string>._, A<GetAatfAeReturnDataCsv>.Ignored))
+                .MustHaveHappened(Repeated.Exactly.Once);
+
+            A.CallTo(() => apiClient.SendAsync(A<string>._, A<GetAatfAeReturnDataCsv>.That.Matches(x => x.AatfDataUrl.Equals("https://localhost:44300/admin/aatf/details/"))));
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ReportsControllerTests.cs]: Need to add .MustHaveHappend on this check

Comment (SumeN) [reply src/EA.Weee.Web.Tests.Unit/Areas/Admin/Controllers/ReportsControllerTests.cs]: Added

**Pull request merged**